pkg/metrics/server_test.go

```diff
@@ -0,0 +1,268 @@
+package metrics
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"testing"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+func init() {
+	key, cert, err := generateTempCertificates()
+	if err != nil {
+		panic(err)
+	}
+	tlsCRT = cert
+	tlsKey = key
+
+	// sets the default http client to skip certificate check.
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
+	}
+}
+
+func generateTempCertificates() (string, string, error) {
+	key, err := rsa.GenerateKey(rand.Reader, 1024)
+	if err != nil {
+		return "", "", err
+	}
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+	}
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, key.Public(), key)
+	if err != nil {
+		return "", "", err
+	}
+
+	cert, err := ioutil.TempFile("/tmp", "testcert-")
```

Comment:
It's better to use `""` instead of `"/tmp"`. And as I can see, after `go test` there will be leftovers in /tmp, can we not pollute /tmp?

Reply:
I have added a fix for this.